cinema/tests/test_movie_api.py

```
@@ -157,3 +156,132 @@ def test_image_url_is_shown_on_movie_session_detail(self):
         res = self.client.get(MOVIE_SESSION_URL)
 
         self.assertIn("movie_image", res.data[0].keys())
+
+
+class AuthenticatedMovieAPITests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="test@mail.com", password="password"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_movies_list(self):
+        sample_movie()
+        movie_with_genre = sample_movie()
+        movie_with_actor = sample_movie()
+
+        genre = sample_genre()
+        actor = sample_actor()
+
+        movie_with_genre.genres.add(genre)
+        movie_with_actor.actors.add(actor)
+
+        res = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_filter_movies_by_genre(self):
+        movie_without_genre = sample_movie()
+        movie_with_genre = sample_movie(title="Title1")
+
+        genre = Genre.objects.create(name="Genre1")
+        movie_with_genre.genres.add(genre)
+
+        res = self.client.get(MOVIE_URL, {"genres": [genre.id]})
+
+        serializer = MovieListSerializer(movie_without_genre)
+        serializer_with_genre = MovieListSerializer(movie_with_genre)
+
+        self.assertIn(serializer_with_genre.data, res.data)
+        self.assertNotIn(serializer, res.data)
```

These assertions compare the serializer object instead of its serialized data. Use `serializer.data` in the negative assertion. Right now the test compares a `MovieListSerializer` instance to `res.data`, which is incorrect and will either always pass or hide a real failure.